include/arch/arc/arch.h

```diff
@@ -79,87 +80,77 @@ extern "C" {
 		1 << (31 - __builtin_clz(x) + 1) : \
 		1 << (31 - __builtin_clz(x)))
 
-#if defined(CONFIG_USERSPACE)
 
+#if defined(CONFIG_USERSPACE)
 #define Z_ARCH_THREAD_STACK_RESERVED \
 	(STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE)
+#else
+#define Z_ARCH_THREAD_STACK_RESERVED (STACK_GUARD_SIZE)
+#endif
 
-#if CONFIG_ARC_MPU_VER == 2
-
-#define Z_ARCH_THREAD_STACK_DEFINE(sym, size) \
-	struct _k_thread_stack_element __noinit \
-		__aligned(POW2_CEIL(STACK_SIZE_ALIGN(size))) \
-		sym[POW2_CEIL(STACK_SIZE_ALIGN(size)) + \
-		+  STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE]
 
+#if  defined(CONFIG_USERSPACE) && CONFIG_ARC_MPU_VER == 2
+/* MPUv2 requires
+ *  - region size must be power of 2 and >= 2048
+ *  - region start must be aligned to its size
+ */
+#define Z_ARCH_THREAD_STACK_SIZE_PWR2(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
+/*
+ * user stack and guard are protected using MPU regions, so need to adhere to
+ * MPU start, size alignment
+ */
+#define Z_ARCH_THREAD_STACK_ALIGN(size)	Z_ARCH_THREAD_STACK_SIZE_PWR2(size)
```

Comment:
Do not introduce new Z_ARCH_ macros, call this something else. Z_ARCV2_ is fine.

Reply:
done